### PR TITLE
feat: integrate sqlmodel, sqlalchemy, alembic, docker

### PR DESCRIPTION
**Pull request by owengretzinger**
### TL;DR

Added database integration with SQLModel, SQLAlchemy, and Alembic. Added Docker support.

Completes MEE-81, MEE-82

### What changed?

- Introduced SQLModel for database ORM with User, Bot, and Event models
- Added Alembic for database migrations
- Created Docker and docker-compose configuration
- Added environment variable support with dotenv
- Implemented CRUD endpoints for users and bots
- Restructured the backend directory for better organization

### How to test?

Follow the updated `README.md`.

### Why make this change?

To establish a proper database layer for storing user, bot, and event data, while making the application containerized and deployable. This provides a foundation for persistent data storage and scalable deployment options.

### CODE REVIEW

**Comment by owengretzinger** [comment]
* **#31** <a href="https://app.graphite.dev/github/pr/meetingbot/meetingbot/31?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/meetingbot/meetingbot/31?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/backend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# pull official base image
+FROM python:3.11-slim-buster
```
Comment (err53) [inline src/backend/Dockerfile:2]: I wonder if there's a way to keep this in sync with the one in pyproject.toml...

Comment (owengretzinger) [reply src/backend/Dockerfile:2]: I mean they're both Python 11, what else can I do?

**Review by sahibkhokhar — APPROVED** [review]

**Comment by owengretzinger** [comment]
### Merge activity

* **Nov 21, 3:01 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/meetingbot/meetingbot/31).

**Pull request merged**